### ISSUE TITLE
Stream.attach_response() returns list. 

### ISSUE THREAD

**Issue by petrrr** [obspy/obspy#1287]
I realized that I get some object back when calling the `Stream.attach_response()` method. When looking up the details I found out that it is a list of `Trace` object. Given that there is a nice collection object for `Traces`, i.e. `Stream`, I am a bit wondering why this could not be returned instead. This would allow to use all the specific functionalities. 

Could this still be changed? `Streams` have quite some of the behavior of lists, right?

**Comment by megies** [comment]
The return value was only intended to check for Traces that could not be looked up, i.e. failed to attach response to.
But.. with #1234 we kinda decided to slowly put the `attach_response` out of service anyway, I guess..

**Comment by petrrr** [comment]
I understand, that the main intention of this feature is to check where the operation fails. Still just having the pretty print functionality of `Stream` sometimes would be nice. 

Anyway, if this is going to go away, probably no need to care ...

**Comment by petrrr** [comment]
BTW: If there is the intention to phase out the `attach_response`, should this be explicitly stated in the docu, even if you might not want to formally deprecate it yet, some hint at the commanded way to implement things might be good. To me it was not very clear why there are these to ways to do the same.

**Comment by megies** [comment]
> some hint at the commanded way to implement things might be good. To me it was not very clear why there are these to ways to do the same.

Yes, some note would not hurt. At least in `remove_response` the example / recommended usage was changed in docstring (https://github.com/obspy/obspy/pull/1234/files#diff-cd1a583debd26f20758184758276d75eR3027)
